Riot/Modules/Common/ActivityPresenters/FullscreenLoadingActivityPresenter.swift

```diff
@@ -0,0 +1,74 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import CommonKit
+import UIKit
+
+/// An `ActivityPresenter` responsible for showing / hiding a full-screen loading view that obscures (and thus disables) all other controls.
+/// It is managed by an `Activity`, meaning the `present` and `dismiss` methods will be called when the parent `Activity` starts or completes.
+class FullscreenLoadingActivityPresenter: ActivityPresentable {
+    private let label: String
+    private weak var viewController: UIViewController?
+    private weak var view: UIView?
+    
+    init(label: String, on viewController: UIViewController) {
+        self.label = label
+        self.viewController = viewController
+    }
+
+    func present() {
+        // Find the current top navigation controller
+        var vc: UIViewController? = viewController
+        while vc?.navigationController != nil {
```

Comment:
SwiftLint warning here `Identifier Name Violation: Variable name should be between 3 and 40 characters long: 'vc' (identifier_name)`

Reply:
You probably need to update .swiftlint with the path to CommonKit. Do me a favor and check the other targets while you're at it.